torch/csrc/autograd/functions/init.cpp

```diff
@@ -62,27 +63,89 @@ struct NoCtor {
 };
 
 template<typename C, typename T>
-static void addClass(PyObject* module, PyTypeObject& type, const char* name)
+static void addClass(PyObject* module, PyTypeObject& type, const char* name,
+  PyGetSetDef* function_properties=NULL, PyMethodDef* function_methods=NULL)
 {
-  createForwardFunctionPyTypeObject<T>(type, name);
+  createForwardFunctionPyTypeObject<T>(type, name, function_properties, function_methods);
   Py_INCREF(&type);
   PyModule_AddObject(module, name, (PyObject*)&type);
   registerCppFunction(typeid(C), &type);
 }
 
+template<typename T, typename ValueT, typename ParamsT, ValueT ParamsT::*ptr,
+         typename ConvertArgT, PyObject* (*Convert)(ConvertArgT)>
+PyObject* getTupleAttr(PyObject* obj, void* _unused)
+{
+  HANDLE_TH_ERRORS
+  THPCppFunction* self = (THPCppFunction*)obj;
+  auto& arr = (T*)(self->cdata.get())->*ptr;
+  auto num_elems = arr.size();
+  THPObjectPtr py_tuple = PyTuple_New(num_elems);
+  if (!py_tuple) return NULL;
+  for (size_t i = 0; i < num_elems; ++i) {
+    PyTuple_SET_ITEM(py_tuple.get(), i, Convert(arr[i]));
+  }
+  return py_tuple.release();
+  END_HANDLE_TH_ERRORS
+}
+
+template<typename T, typename ValueT, typename ParamsT, ValueT ParamsT::*ptr,
+         typename ConvertArgT, PyObject* (*Convert)(ConvertArgT)>
+PyObject* getValueAttr(PyObject* obj, void* _unused)
+{
+  HANDLE_TH_ERRORS
+  THPCppFunction* self = (THPCppFunction*)obj;
+  auto& val = (T*)(self->cdata.get())->*ptr;
+  return Convert(val);
+  END_HANDLE_TH_ERRORS
+}
+
+static struct PyGetSetDef conv_forward_properties[] = {
+  THP_FUNCTION_DEFAULT_PROPERTIES,
+  {(char*)"stride", (getter)getTupleAttr<ConvForward, std::vector<int>, ConvParams, 
+                                         &ConvParams::stride, long, PyInt_FromLong>, NULL, NULL, NULL},
+  {(char*)"padding", (getter)getTupleAttr<ConvForward, std::vector<int>, ConvParams, 
+                                         &ConvParams::padding, long, PyInt_FromLong>, NULL, NULL, NULL},
+  {(char*)"dilation", (getter)getTupleAttr<ConvForward, std::vector<int>, ConvParams, 
+                                         &ConvParams::dilation, long, PyInt_FromLong>, NULL, NULL, NULL},
+  {(char*)"transposed", (getter)getValueAttr<ConvForward, bool, ConvParams, 
+                                         &ConvParams::transposed, long, PyBool_FromLong>, NULL, NULL, NULL},
+  {(char*)"output_padding", (getter)getTupleAttr<ConvForward, std::vector<int>, ConvParams, 
+                                         &ConvParams::output_padding, long, PyInt_FromLong>, NULL, NULL, NULL},
```

Comment:
More details on the above: removing the `ParamsT` template parameter, since `T::*ptr` should suffice, namely
```
template<typename T, typename ValueT, ValueT T::*ptr,
         typename ConvertArgT, PyObject* (*Convert)(ConvertArgT)>
PyObject* getTupleAttr(PyObject* obj, void* _unused)
{
  HANDLE_TH_ERRORS
  THPCppFunction* self = (THPCppFunction*)obj;
  auto& arr = ((T*)(self->cdata.get()))->*ptr;
  auto num_elems = arr.size();
  THPObjectPtr py_tuple = PyTuple_New(num_elems);
  if (!py_tuple) return NULL;
  for (size_t i = 0; i < num_elems; ++i) {
    PyTuple_SET_ITEM(py_tuple.get(), i, Convert(arr[i]));
  }
  return py_tuple.release();
  END_HANDLE_TH_ERRORS
}

template<typename T, typename ValueT, ValueT T::*ptr,
         typename ConvertArgT, PyObject* (*Convert)(ConvertArgT)>
PyObject* getValueAttr(PyObject* obj, void* _unused)
{
  HANDLE_TH_ERRORS
  THPCppFunction* self = (THPCppFunction*)obj;
  auto& val = ((T*)(self->cdata.get()))->*ptr;
  return Convert(val);
  END_HANDLE_TH_ERRORS
}
```
and using `&ConvForward::output_padding` (it should be valid since `ConvForward` inherits from `ConvParams` and from `Function`) 
```
{(char*)"output_padding", (getter)getTupleAttr<ConvForward, std::vector<int>,
                                         &ConvForward::output_padding, long, PyInt_FromLong>, NULL, NULL, NULL},
```
I get the following error (macOS, clang-802.0.42):
```
torch/csrc/autograd/functions/init.cpp:113:37: error: address of overloaded function 'getTupleAttr' does not match required type
      '_object *(_object *, void *)'
  {(char*)"output_padding", (getter)getTupleAttr<ConvForward, std::vector<int>,
                                    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
torch/csrc/autograd/functions/init.cpp:77:11: note: candidate template ignored: invalid explicitly-specified argument for template
      parameter 'ptr'
PyObject* getTupleAttr(PyObject* obj, void* _unused)
          ^
```
Things go smoothly if I provide `output_padding` as a member of `ConvParams`.

Definitely not a roadblock, but making the above work would be a bit cleaner.